Day2/ex2.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+
+type Rating struct { // Struct for rating
+	TotalRating float64
+	mutex sync.Mutex
+}
+
+type Student struct { // Struct for student
+	id int
+}
+
+func (student Student) getrating(wg *sync.WaitGroup, totalrating *Rating) { // Function for getting rating from a student
+	defer wg.Done()
+	time.Sleep(time.Millisecond * 50)
+	totalrating.mutex.Lock()
+	totalrating.TotalRating += 0.01 * float64(rand.Intn(1000))
+	totalrating.mutex.Unlock()
+}
+
+func main() {
+	rand.Seed(time.Now().UnixNano()) // Changing seed according to time
+	const NumberOfStudents = 200
+	var totalrating = Rating{0.0, sync.Mutex{}}
+	Students := []Student{}
```

Comment:
using var is more idiomatic

```suggestion
	var Students []Student
```